cmd/object-handlers.go

```diff
@@ -643,7 +643,15 @@ func getCpObjMetadataFromHeader(ctx context.Context, r *http.Request, userMeta m
 	// if x-amz-metadata-directive says REPLACE then
 	// we extract metadata from the input headers.
 	if isMetadataReplace(r.Header) {
-		return extractMetadata(ctx, r)
+		meta, err := extractMetadata(ctx, r)
+		if err != nil {
+			return meta, err
+		}
+		// Set content-type to default value if it is not set.
+		if _, ok := meta["Content-Type"]; !ok {
```

Comment:
move this check inside `extractMetadata` and `Content-Type` is actually lower case here @kannappanr that's why its not working.

Reply:
done